credoai/artifacts/model/classification_model.py

```diff
@@ -117,11 +123,16 @@ def __post_init__(self):
 
         elif self.model_info["framework"] == "credoai":
             # Functionality for DummyClassifier
-            self.model_like = getattr(self.model_like, "model_like", None)
+            if self.model_like.model_like is not None:
+                self.model_like = self.model_like.model_like
             # If the dummy model has a model_like specified, reassign
             # the classifier's model_like attribute to match the dummy's
             # so that downstream evaluators (ModelProfiler) can use it
 
+            self.type = self.model_like.type
```

Comment:
Do we need to do it like this? Can't we just count the number of unique values that are created by the predict function?

Or perhaps we can do this to automatically determine the type within DummyClassifier...

Reply:
That occurred to me but it's possible the supplied predictions don't actually contain samples with labels for all possible classes and thus we might incorrectly infer that it's supposed to be a binary problem when it's really multi-class.